DDSaaS: StreamNative: Crawler Integration v1.0.0 (#18782)
DataDog/integrations-core: streamnative/README.md

# StreamNative

## Overview

[StreamNative][1] provides an enterprise-grade messaging and event streaming platform built on Apache Pulsar. It offers scalable, real-time data streaming solutions with features like multi-tenancy, geo-replication, and seamless integration with cloud services.

> The StreamNative integration collects the below types of [metrics][2]:

Reviewer: "collects the below" > Collects the following types of metrics

Author: Done.

1. Health
2. Pulsar Resource
3. Source Connector
4. Sink Connector
5. Kafka Connect

## Setup

### Configuration

#### Get StreamNative credentials

##### Log in to the [StreamNative Cloud Console Account][3].
##### Get the `Organization ID` and `Instance Name`:
1. Click the profile icon and select the **Organizations** option.
2. Select the **Organization** for which data needs to be collected.
3. Obtain the **Instance Name** from the **Select an Instance** dropdown.
##### Get the `Client ID` and `Client Secret`:
1. Click the profile icon and select the **Accounts & Accesses** tab.
2. Select the Service Account for which **Admin** permission is **Enabled**.
3. If the Service Account does not exist, click on **New -> Service Account** to create a new one, and ensure the **Super Admin** option is enabled.
4. Click on the `...` button on the right side of the selected Service Account.
5. Click **Download OAuth2 Key** to get the **Client ID** and **Client Secret**.

> #### Add StreamNative credentials

Reviewer: Can we please get more clarification here? "Add the following Stream Native Credentials to the Datadog Tile"

Reviewer 2: Hi @apiazza-dd, we have update the first half including the steps to get the configuration details. 
Regarding the second part from "Setting Up the Integration Tile", earlier on the similar lines we have had an discussion around this over slack with @dpoint01 ( [ref](https://crestdatasystems.slack.com/archives/C079S5829S9/p1729086353361489) ). Can you please have a look. thanks!

Reviewer: @ashah-crest Thanks, I don't have access to that slack environment, but if @dpoint01  approved, lgtm.

# StreamNative

## Overview

[StreamNative][1] provides an enterprise-grade messaging and event streaming platform built on Apache Pulsar. It offers scalable, real-time data streaming solutions with features like multi-tenancy, geo-replication, and seamless integration with cloud services.

> The StreamNative integration Collects the following types of [metrics][2]:

Reviewer 3: ```suggestion
The StreamNative integration collects the following types of [metrics][2]:
```

Author: Done.

1. Health
2. Pulsar Resource
3. Source Connector
4. Sink Connector
5. Kafka Connect

## Setup

### Configuration

#### Get StreamNative credentials

> ##### Log in to the [StreamNative Cloud Console Account][3].

Reviewer 3: ```suggestion

Log into the [StreamNative Cloud Console Account][3].

```

Author: Done.

# StreamNative

## Overview

[StreamNative][1] provides an enterprise-grade messaging and event streaming platform built on Apache Pulsar. It offers scalable, real-time data streaming solutions with features like multi-tenancy, geo-replication, and seamless integration with cloud services.

> The StreamNative integration collects below types of [metrics][2]:

Reviewer 4: ```suggestion
The StreamNative integration collects the below types of [metrics][2]:
```

Reviewer 4: Or just "collects the below metrics"

Author: Done.

1. Health
2. Pulsar Resource
3. Source Connector
4. Sink Connector
5. Kafka Connect

## Setup

### Configuration

> #### Get StreamNative Credentials

Author: Done.

> ##### Login to the [StreamNative Cloud Console Account][3].

Reviewer 4: ```suggestion
##### Log in to the [StreamNative Cloud Console Account][3].
```

Author: Done.

> 2. Get the `Organization ID` and `Instance Name`

Reviewer 4: ```suggestion
2. Get the `Organization ID` and `Instance Name`:
```

Reviewer 4: You might consider making these top-level numbered items H5s instead, so the list won't be quite so long. I also recommend using a numbered list when the items need to be completed in order, rather than the bullets used below.

Author: Done.

    - Click the profile icon and select the **Organizations** option.
    - Select the **Organization** for which data needs to be collected.
    - Obtain the **Instance Name** from the **Select an Instance** dropdown.

> 3. Get the `Client ID` and `Client Secret`

Reviewer 4: ```suggestion
3. Get the `Client ID` and `Client Secret`:
```

Author: Done.

    - Click the profile icon and select the **Accounts & Accesses** tab.
    - Select the Service Account for which **Admin** permission is **Enabled**.
    - If the Service Account does not exist, click on **New -> Service Account** to create a new one, and ensure the **Super Admin** option is enabled.

>     - Click on `...` button on the right side of the selected Service Account.

Reviewer 4: ```suggestion
    - Click on the `...` button on the right side of the selected Service Account.
```

Author: Done.

5. Click **Download OAuth2 Key** to get the **Client ID** and **Client Secret**.

> #### Add StreamNative Credentials

Reviewer 4: ```suggestion
#### Add StreamNative credentials
```

Author: Done.

> #### Add StreamNative Credentials

Reviewer 4: Where is the user adding these?

Author: On Datadog Integration tile.

Reviewer 2: Hi @jhgilbert , earlied we have had an discussion around this over slack with @dpoint01 ( [ref](https://crestdatasystems.slack.com/archives/C079S5829S9/p1729086353361489) ). Can you please have a look .

Reviewer 4: I'm not able to see the ref thread, but I trust you if you prefer to leave the content as is. I'll resolve the conversation but feel free to reopen it if you want to discuss it further!

- Organization ID 
- Instance Name
- Client ID  
- Client Secret  

> ## Data Collected

Reviewer: We can automatically populate the data collected tab using the following: 

```
## Data Collected

### Metrics

{{< get-metrics-from-git "streamnative" >}}
```

Author: Done.

### Logs 

The StreamNative integration does not include any logs.

### Metrics

The StreamNative integration collects and forwards the following metrics to Datadog.

> 1. Health

Reviewer 5: @apiazza-dd is this where we should have the table view with all the metric names + descriptions for the public docs?

Reviewer 5: The Metrics section on the public docs should reflect a table view of all the metrics collected that show up on the "Data collected" tab on the tile (from the metadata.csv)

# StreamNative

## Overview

[StreamNative][1] provides an enterprise-grade messaging and event streaming platform built on Apache Pulsar. It offers scalable, real-time data streaming solutions with features like multi-tenancy, geo-replication, and seamless integration with cloud services.

> The StreamNative integration collects below types of metrics:

Reviewer: Can we please get a link to the types of metrics collected?

Author: Done.

1. Health
2. Pulsar Resource
3. Source Connector
4. Sink Connector
5. Kafka Connect

## Setup

### Configuration

> StreamNative integration requires a StreamNative account and its Client Id, Client Secret, Organization ID and Instance Name. Below are the steps to fetch these details from StreamNative console:

Reviewer: Id > ID for consistency please.

Author: Done.

#### Steps to retrieve Organization ID, Instance Name, Client Id & Client Secret

> 1. Login to the [StreamNative Cloud Console Account][2] 

Reviewer: Please add a `.` at the end of #1 for consistency. Thanks!

2. Go to the profile icon and select to the **Organizations** option.

> 3. Go into the **Organization** from which data needs to be collected in the Datadog.

Reviewer: the Datadog > Datadog

4. Copy and save the **Organization ID** to configure in the Integration.

> 5. From the **Select an Instance** section, Copy and save the **Instance Name** to configure in the Integration.

Reviewer: Copy > copy

> 6. Go to the profile icon and select the **Accounts & Accesses** tab.

Reviewer: Go to > C"click"

> 7. Choose the Service Account for which **Admin** permission is **True**.

Reviewer: Admin permission is True > `Admin permission is enabled`

Author: Done.

8. If the Service Account does not exist, click on **New -> Service Account** to create a new one, and ensure the **Super Admin** type is enabled.
9. Click on the 3 dots on the right side of the selected Service Account.
10. Click on **Download OAuth2 Key**, this file will have required **Client ID** & **Client Secret** to configure in the Integration.

#### StreamNative DataDog integration configuration

Configure the Datadog endpoint to forward StreamNative metrics to Datadog.

1. Navigate to the `StreamNative` integration tile in Datadog.
2. Add your StreamNative credentials.

| StreamNative parameters | Description                                    |
| ------------------------------ | ---------------------------------------------  |
| Organization ID                   | Organization ID of your StreamNative account.     |
| Instance Name                   | Instance name of the desired organization.     |
| Client ID                   | Client ID of your service account.     |
| Client Secret                   | Client Secret of your service account.     |

## Data Collected

### Logs 

The StreamNative integration does not include any logs.

### Metrics

> The StreamNative integration collects and forward below metrics to Datadog.

Reviewer: The StreamNative integration collects and forwards the following metrics to Datadog.

Author: Done.
